src/Compiler/pars.fsy

```diff
@@ -2588,7 +2588,7 @@ intersectionConstraints:
 
         ($3 :: constraints), (rhs parseState 2 :: mAmpersands) }
 
-  | atomType
+  | hashConstraint
       { match $1 with
         | SynType.HashConstraint _ -> ()
         | ty -> errorR(Error(FSComp.SR.parsConstraintIntersectionSyntaxUsedWithNonFlexibleType(), ty.Range))
```

Comment:
Does this just need to become something like

```fsharp
| hashConstraint
    { [ $1 ], [] }

| atomType
    { errorR(Error(FSComp.SR.parsConstraintIntersectionSyntaxUsedWithNonFlexibleType(), $1.Range))
      [ $1 ], [] }
```

?

Reply:
Yes, thanks!